sonar-text-plugin/src/test/java/org/sonar/plugins/secrets/utils/UpdatingSpecificationFilesGenerator.java

```diff
@@ -59,11 +66,17 @@ class UpdatingSpecificationFilesGenerator {
   private final Charset charset = StandardCharsets.UTF_8;
   private static final Logger LOG = LoggerFactory.getLogger(UpdatingSpecificationFilesGenerator.class);
 
+  @RegisterExtension
+  LogTesterJUnit5 logTester = new LogTesterJUnit5();
+
   // Suppress warning, as there are no assertions inside here
   @Test
   @SuppressWarnings("java:S2699")
   void firstStep() {
     writeSpecificationFileDefinition();
+    assertThatNoException().isThrownBy(SpecificationLoader::new);
+    List<LogAndArguments> errorLogs = logTester.getLogs(Level.ERROR);
+    assertThat(errorLogs).withFailMessage(() -> failMessage(errorLogs)).isEmpty();
```

Comment:
I would suggest to avoid test assertions here. As it is only a workaround to put generation logic in a test, I think it may cause problems if we rely on AssertJ here if (or when) we move this codegen elsewhere. Simply checking condition and throwing a RuntimeException with a message would be enough IMO